### PR TITLE
[v11] Backport of dependabot CVE updates

### PR DESCRIPTION
**Pull request by jentfoo**
Cherry-picked security updates that have been applied in master but failed to backport to v11

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -19,10 +19,11 @@ require (
 	go.opentelemetry.io/otel/sdk v1.10.0
 	go.opentelemetry.io/otel/trace v1.10.0
 	go.opentelemetry.io/proto/otlp v0.19.0
-	golang.org/x/crypto v0.0.0-20220926161630-eccd6366d1be
-	golang.org/x/exp v0.0.0-20230105202349-8879d0199aa3
-	golang.org/x/net v0.0.0-20220927171203-f486391704dc
-	google.golang.org/grpc v1.49.0
+	golang.org/x/crypto v0.2.0 // replaced
```
[the remark this answers is not in the document]
Comment (jentfoo) [reply api/go.mod:22]: As part of the cherry-picking it looks like the replacement was removed.  My plan is to clean these details up in an additional commit, so I will remove the comment as part of that.  Thank you!

File: go.mod [hunk 2]
```diff
@@ -4,8 +4,8 @@ go 1.19
 
 require (
 	cloud.google.com/go/container v1.7.0
-	cloud.google.com/go/firestore v1.6.1
-	cloud.google.com/go/iam v0.6.0
+	cloud.google.com/go/firestore v1.9.0
```
[the remark this answers is not in the document]
Comment (codingllama) [reply go.mod:7]: Maybe move Firestore to a separate PR? It's a bigger and more delicate change than the others.

[the remark this answers is not in the document]
Comment (jentfoo) [reply go.mod:7]: I believe this change can be reverted.  It came in as part of the dependabot-batcher PR, but is not related to a CVE

[the remark this answers is not in the document]
Comment (zmb3) [reply go.mod:7]: +1

[the remark this answers is not in the document]
Comment (jentfoo) [reply go.mod]: Reverted, there are still a good amount of other updates included that are not CVE related.  I am seeing if the build passes, but feel free to call out any others that seem potentially risky.  Thank you!

[the remark this answers is not in the document]
Comment (jentfoo) [reply go.mod]: After the unit test failures I agree.  Let's land the already minimal #23577 and use that as a reference for what needs updated in here and v12

[the remark this answers is not in the document]
Comment (codingllama) [reply go.mod]: Your call. Not every change on release branches needs to be a backport, so feel free to start this from a clean slate if that's better.

**Comment by jentfoo** [comment]
@codingllama, @rosstimothy, @zmb3, this PR has now been reduced to a more minimal set of updates.  Can you please review this?  Thank you!

File: assets/backport/go.mod [hunk 3]
```diff
@@ -17,10 +17,10 @@ require (
...
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/sirupsen/logrus v1.8.1 // indirect
-	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 // indirect
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110 // indirect
-	golang.org/x/sys v0.0.0-20210615035016-665e8c7367d1 // indirect
-	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
+	golang.org/x/crypto v0.1.0 // indirect
+	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	golang.org/x/term v0.5.0 // indirect
```
[the remark this answers is not in the document]
Comment (jentfoo) [reply assets/backport/go.mod:23]: Typically I would, but with the goal of keeping these updates minimal I am not currently.  I suspect that these API's are stable enough that this should be the safest route, but if there is a preference for a greater change I can update the root dependencies depending on these.

File: assets/backport/go.mod [hunk 4]
```diff
@@ -17,10 +17,10 @@ require (
 	github.com/jonboulle/clockwork v0.2.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/sirupsen/logrus v1.8.1 // indirect
-	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 // indirect
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110 // indirect
-	golang.org/x/sys v0.0.0-20210615035016-665e8c7367d1 // indirect
-	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
+	golang.org/x/crypto v0.1.0 // indirect
```
Comment (codingllama) [inline assets/backport/go.mod:20]: Bump to 0.2.0 while we're at it?

Should we do the same for web/.cloudbuild/scripts/go.mod as well?

**Review by codingllama — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**